backend/infrahub/core/relationship/model.py

```diff
@@ -1110,7 +1110,7 @@ async def remove_in_db(
         # - Update the existing relationship if we are on the same branch
         rel_ids_per_branch = peer_data.rel_ids_per_branch()
         if branch.name in rel_ids_per_branch:
-            await update_relationships_to([str(ri) for ri in rel_ids_per_branch[self.branch.name]], to=remove_at, db=db)
+            await update_relationships_to([str(ri) for ri in rel_ids_per_branch[branch.name]], to=remove_at, db=db)
```

Comment:
nasty one

Reply:
I think it's confusing that `self.branch` ends up being `main` here